utils/validate_images.ts

```diff
@@ -1,106 +1,68 @@
-  getFileExtension,
-  globFiles,
-  readQuickstartFile,
-  findMainQuickstartConfigFiles,
-} = require('./helpers');
-
-const glob = require('glob');
-const path = require('path');
-const BASE_PATH = '../quickstarts/';
-const DASHBOARD_IMAGES_PATH = '/images/';
-const MAX_SIZE = 4000000;
-const MAX_NUM_IMG = 12;
-const ALLOWED_IMG_EXT = ['.png', '.jpeg', '.jpg', '.svg'];
+import * as core from '@actions/core';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as fs from 'fs';
+import Dashboard from './lib/Dashboard';
 
 type DirectoryValidation = {
   folder: string;
-  dashboardCount: number;
   imageCount: number;
   maxImages: number;
 };
 
+const BASE_PATH = '../dashboards/';
+const MAX_SIZE = 4000000;
+const MAX_NUM_IMG = 12;
+const ALLOWED_IMG_EXT = ['.png', '.jpeg', '.jpg', '.svg'];
+const IGNORE_FILE_EXT = ['.json']
+
 /**
- * Validate all folders contain no more than MAX_NUM_IMG images
+ * Gets the size of a file in Bytes
+ * @param filePath - The file to get the size of.skip
+ * @returns - The file size in Bytes
  */
-export const validateImageCounts = (quickstartDirs: string[]): void => {
-  const screenshotDirectories: DirectoryValidation[] = [];
-  const imagesDirectories: DirectoryValidation[] = [];
-
-  quickstartDirs.forEach((quickstart: string) => {
-    const quickstartDirName = path.dirname(quickstart);
-    // get all images for a quickstart
-    const imagePaths = glob.sync(
-      path.resolve(quickstartDirName, 'dashboards/**/*.+(png|jpeg|jpg|svg)')
-    );
-    const quickstartConfig = readQuickstartFile(quickstart).contents[0];
-    const quickstartName = quickstartConfig.name;
-    const logoPath = quickstartConfig.logo
-      ? path.resolve(quickstartDirName, quickstartConfig.logo)
-      : null;
-
-    // Max images is per dashboard so we need to account for this by getting the number of dashboards
-    const dashboardCount = glob.sync(
-      path.resolve(quickstartDirName, 'dashboards/**/*.json')
-    ).length;
+export const getFileSize = (filePath: string): number => {
+  return fs.statSync(filePath)['size'];
+};
 
-    const screenshotPaths = imagePaths.filter(
-      (p: string[]) =>
-        p !== logoPath && !p.includes(quickstartName + DASHBOARD_IMAGES_PATH)
-    );
+/**
+ * Validate all dashboard folders contain no more than MAX_NUM_IMG images
+ */
+export const validateImageCounts = (dashboards: Dashboard[]): void => {
+  const imagesDirectories: DirectoryValidation[] = [];
+  dashboards.forEach((dashboard) => {
+    const dashboardDirName = path.dirname(dashboard.configPath);
 
-    const dashboardImagePaths = imagePaths.filter(
-      (p: string[]) =>
-        p !== logoPath && p.includes(quickstartName + DASHBOARD_IMAGES_PATH)
-    );
+    const dashboardImagePaths = glob
+      .sync(path.join('..', dashboardDirName))
+      .filter((filePath) => ALLOWED_IMG_EXT.includes(path.extname(filePath)))
+      .filter((filePath) => IGNORE_FILE_EXT.includes(path.extname(filePath)));
```

Comment:
We might be missing something here, but this appears to filter down to just image files, then filter again down to just JSON files, which should leave us with an empty array. 
If we're correct, then we can update it to:
```suggestion
      .sync(path.join('..', dashboardDirName))
      .filter((filePath) => ALLOWED_IMG_EXT.includes(path.extname(filePath)));
```

Reply:
This is resolved - we now filter out any JSON files before being sent to the validation functions